### PR TITLE
drivers: i2c_ll_stm32_v1: Fix TX and RX operations

### PR DESCRIPTION
**Pull request by ydamigos**
Fixes single byte reception. In case a single byte has to be received, the Acknowledge
disable is made before ADDR flag is cleared.

Disables/enables only Buffer interrupts. Use the LL_I2C_EnableIT_BUF and  LL_I2C_DisableIT_BUF functions to enable and disable the I2C Buffer interrupts (TxE, RxNE). The LL_I2C_*IT_TX and LL_I2C_*IT_RX functions enable and disable both I2C Buffer and Event interrupts.

Fixes TX if flag I2C_MSG_RESTART is set. In interrupt mode, the drivers entered a forever loop if the
I2C_MSG_RESTART flag was set.

Fixes RX operation for all the receptions (1-byte, 2-bytes, N-bytes when N > 2).

Both interrupt and polling modes were tested on 96b_carbon.

### CODE REVIEW

**Comment by ydamigos** [comment]
@ldts Could you please check it.

**Comment by Inakimalerba** [comment]
+1 Tested with 96b_carbon and fxos8700

**Review by galak — CHANGES_REQUESTED** [review]
For the first to commits, can you add more detail on the 'why' and 'how' you fix the issues in the commit messages.

**Comment by ydamigos** [comment]
@galak I squashed the first commit with the last, as it was actually part of handling 1-byte reception and I updated the second commit message.

File: drivers/i2c/i2c_ll_stm32_v1.c [hunk 1]
```diff
@@ -72,13 +72,24 @@ static inline void handle_addr(I2C_TypeDef *i2c, struct i2c_stm32_data *data)
...
+			 * This is the last byte to transmit disable Buffer
+			 * interrupt and wait for a BTF interrupt
+			 */
+			LL_I2C_DisableIT_BUF(i2c);
+		}
 		LL_I2C_TransmitData8(i2c, *data->current.buf);
 		data->current.buf++;
-		data->current.len--;
-	} else if (LL_I2C_IsActiveFlag_BTF(i2c) && !data->current.len) {
+	} else if (!data->current.len) {
```
Comment (galak) [inline drivers/i2c/i2c_ll_stm32_v1.c]: do we really need the if part(), is the else not sufficient?

Comment (ydamigos) [inline drivers/i2c/i2c_ll_stm32_v1.c]: Yes, else is enough.

**Review by galak — APPROVED** [review]

**Pull request merged**